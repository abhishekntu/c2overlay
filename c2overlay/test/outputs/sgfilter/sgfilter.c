/****************************************************************/
/*   chebyshev		                                        */
/****************************************************************/

#ifdef TEST
#include <stdio.h>
#endif

#include<math.h>

#ifdef ALGO
int foo(int x, int y)
{
 	return (x*(x*(7*x - 76*y +7) + y*(92*y - 39) +7) - y*(y*(984*y + 46) + 46) -75);	
}
#endif


#ifdef TEST
int main(void)
{
	int x,y;
 	int result;
  	result = foo(x,y);
  	return 0;
}
#endif
