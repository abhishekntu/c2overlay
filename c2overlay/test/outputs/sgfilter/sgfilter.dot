digraph graphname {
N12 [color="black", label="N7N6_mul_N7_add_N6_N12"];
N13 [color="black", label="O0_N21_N13"];
N10 [color="black", label="N3N8_add_Imm_7_N3_mul_N8_N10"];
N11 [color="black", label="N5N14_add_Imm_7_N5_mul_N14_N11"];
N8 [color="black", label="N15N18_mul_Imm_-984_N15_sub_Imm_46_N18_N8"];
N9 [color="black", label="N9N20_mul_Imm_92_N9_sub_Imm_39_N20_N9"];
N1 [color="black", label="I0_N1_N1"];
N2 [color="black", label="I1_N2_N2"];
N3 [color="black", label="N12N19_mul_N12_sub_Imm_46_N19_N3"];
N4 [color="black", label="N13N16_add_N13_sub_Imm_75_N16_N4"];
N5 [color="black", label="N10N4_mul_Imm_7_N10_add_N4_N5"];
N6 [color="black", label="mul_Imm_-76_N11_N6"];
N7 [color="black", label="mul_N17_N7"];
N12 -> N11;
N10 -> N12;
N11 -> N4;
N8 -> N3;
N9 -> N12;
N1 -> N12;
N1 -> N3;
N1 -> N9;
N1 -> N7;
N1 -> N8;
N1 -> N6;
N2 -> N5;
N2 -> N10;
N2 -> N11;
N3 -> N7;
N4 -> N13;
N5 -> N10;
N6 -> N5;
N7 -> N4;
}
