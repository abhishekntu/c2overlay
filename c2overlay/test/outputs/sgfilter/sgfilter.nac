procedure foo(in s32 x,in s32 y,out s32 D_2404)
{
  localvar s32 D_2405;
  localvar s32 D_2406;
  localvar s32 D_2407;
  localvar s32 D_2408;
  localvar s32 D_2409;
  localvar s32 D_2410;
  localvar s32 D_2411;
  localvar s32 D_2412;
  localvar s32 D_2413;
  localvar s32 D_2414;
  localvar s32 D_2415;
  localvar s32 D_2416;
  localvar s32 D_2417;
  localvar s32 D_2418;
  localvar s32 D_2419;
  localvar s32 D_2420;
  localvar s32 D_2421;
L0005:
  D_2405 <= mul x,7;
  D_2406 <= mul y,-76;
  D_2407 <= add D_2405,D_2406;
  D_2408 <= add D_2407,7;
  D_2409 <= mul D_2408,x;
  D_2410 <= mul y,92;
  D_2411 <= sub D_2410,39;
  D_2412 <= mul D_2411,y;
  D_2413 <= add D_2409,D_2412;
  D_2414 <= add D_2413,7;
  D_2415 <= mul D_2414,x;
  D_2416 <= mul y,-984;
  D_2417 <= sub D_2416,46;
  D_2418 <= mul D_2417,y;
  D_2419 <= sub D_2418,46;
  D_2420 <= mul D_2419,y;
  D_2421 <= add D_2415,D_2420;
  D_2404 <= sub D_2421,75;
}
