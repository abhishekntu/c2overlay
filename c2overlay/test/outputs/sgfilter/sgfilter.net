.input N1
pinlist: N1
    
.input N2
pinlist: N2
    
.output out:N13
pinlist: N13
    
.clb N12_blk
pinlist: N9 N1 N10 open N12 open open open open 
subblock: N12_blk 0 1 2 open 4 open open open open
    
.clb N10_blk
pinlist: N5 N2 open open N10 open open open open 
subblock: N10_blk 0 1 open open 4 open open open open

.clb N11_blk
pinlist: N2 N12 open open N11 open open open open 
subblock: N11_blk 0 1 open open 4 open open open open

.clb N8_blk
pinlist: N1 open open open N8 open open open open 
subblock: N8_blk 0 open open open 4 open open open open

.clb N9_blk
pinlist: N1 open open open N9 open open open open 
subblock: N9_blk 0 open open open 4 open open open open

.clb N3_blk
pinlist: N8 N1 open open N3 open open open open 
subblock: N3_blk 0 1 open open 4 open open open open

.clb N4_blk
pinlist: N7 N11 open open N13 open open open open 
subblock: N4_blk 0 1 open open 4 open open open open

.clb N5_blk
pinlist: N6 N2 open open N5 open open open open 
subblock: N5_blk 0 1 open open 4 open open open open

.clb N6_blk
pinlist: N1 open open open N6 open open open open 
subblock: N6_blk 0 open open open 4 open open open open

.clb N7_blk
pinlist: N3 N1 open open N7 open open open open 
subblock: N7_blk 0 1 open open 4 open open open open

