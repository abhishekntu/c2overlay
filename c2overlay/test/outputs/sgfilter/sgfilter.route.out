Array size: 5 x 5 logic blocks.

Routing:

Net 0 (N1)

SOURCE (6,2)  Pad: 1  
  OPIN (6,2)  Pad: 1  
 CHANY (5,2)  Track: 1  
  IPIN (5,2)  Pin: 1  
  SINK (5,2)  Class: 0  
 CHANY (5,2)  Track: 1  
 CHANX (5,1)  Track: 1  
 CHANX (4,1)  Track: 1  
  IPIN (4,2)  Pin: 2  
  SINK (4,2)  Class: 0  
 CHANX (4,1)  Track: 1  
 CHANX (3,1)  Track: 1  
  IPIN (3,2)  Pin: 2  
  SINK (3,2)  Class: 0  
 CHANX (3,1)  Track: 1  
 CHANX (2,1)  Track: 1  
  IPIN (2,2)  Pin: 2  
  SINK (2,2)  Class: 0  
 CHANX (2,1)  Track: 1  
 CHANX (1,1)  Track: 1  
  IPIN (1,2)  Pin: 2  
  SINK (1,2)  Class: 0  
 CHANX (1,1)  Track: 1  
  IPIN (1,1)  Pin: 0  
  SINK (1,1)  Class: 0  


Net 1 (N2)

SOURCE (6,1)  Pad: 1  
  OPIN (6,1)  Pad: 1  
 CHANY (5,1)  Track: 1  
  IPIN (5,1)  Pin: 1  
  SINK (5,1)  Class: 0  
 CHANY (5,1)  Track: 1  
 CHANX (5,0)  Track: 1  
 CHANX (4,0)  Track: 1  
  IPIN (4,1)  Pin: 2  
  SINK (4,1)  Class: 0  
 CHANX (4,0)  Track: 1  
 CHANX (3,0)  Track: 1  
  IPIN (3,1)  Pin: 2  
  SINK (3,1)  Class: 0  


Net 2 (N12)

SOURCE (4,2)  Class: 1  
  OPIN (4,2)  Pin: 6  
 CHANX (4,1)  Track: 0  
  IPIN (4,1)  Pin: 0  
  SINK (4,1)  Class: 0  


Net 3 (N10)

SOURCE (4,1)  Class: 1  
  OPIN (4,1)  Pin: 7  
 CHANY (3,1)  Track: 1  
  IPIN (3,1)  Pin: 1  
  SINK (3,1)  Class: 0  


Net 4 (N11)

SOURCE (3,1)  Class: 1  
  OPIN (3,1)  Pin: 7  
 CHANY (2,1)  Track: 1  
  IPIN (2,1)  Pin: 1  
  SINK (2,1)  Class: 0  


Net 5 (N8)

SOURCE (1,1)  Class: 1  
  OPIN (1,1)  Pin: 4  
 CHANX (1,1)  Track: 0  
 CHANY (1,2)  Track: 0  
  IPIN (1,2)  Pin: 1  
  SINK (1,2)  Class: 0  


Net 6 (N9)

SOURCE (3,2)  Class: 1  
  OPIN (3,2)  Pin: 5  
 CHANY (3,2)  Track: 1  
  IPIN (4,2)  Pin: 3  
  SINK (4,2)  Class: 0  


Net 7 (N3)

SOURCE (1,2)  Class: 1  
  OPIN (1,2)  Pin: 5  
 CHANY (1,2)  Track: 1  
  IPIN (2,2)  Pin: 3  
  SINK (2,2)  Class: 0  


Net 8 (N4)

SOURCE (5,1)  Class: 1  
  OPIN (5,1)  Pin: 7  
 CHANY (4,1)  Track: 1  
  IPIN (4,1)  Pin: 1  
  SINK (4,1)  Class: 0  


Net 9 (N5)

SOURCE (5,2)  Class: 1  
  OPIN (5,2)  Pin: 6  
 CHANX (5,1)  Track: 0  
  IPIN (5,1)  Pin: 0  
  SINK (5,1)  Class: 0  


Net 10 (N13)

SOURCE (2,1)  Class: 1  
  OPIN (2,1)  Pin: 6  
 CHANX (2,0)  Track: 1  
  IPIN (2,0)  Pad: 0  
  SINK (2,0)  Pad: 0  


Net 11 (N7)

SOURCE (2,2)  Class: 1  
  OPIN (2,2)  Pin: 5  
 CHANY (2,2)  Track: 1  
 CHANX (3,1)  Track: 0  
  IPIN (3,1)  Pin: 0  
  SINK (3,1)  Class: 0  
